code/modules/paperwork/paperbin.dm

```diff
@@ -15,30 +16,58 @@
 	var/total_paper = 30
 	var/list/papers = list()
 	var/obj/item/pen/bin_pen
+	///Overlay of the pen on top of the bin.
+	var/mutable_appearance/pen_overlay
+	///Name of icon that goes over the paper overlays.
+	var/bin_overlay_string = "paper_bin_overlay"
+	///Overlay that goes over the paper overlays.
+	var/mutable_appearance/bin_overlay
 
 /obj/item/paper_bin/Initialize(mapload)
 	. = ..()
 	interaction_flags_item &= ~INTERACT_ITEM_ATTACK_HAND_PICKUP
 	AddElement(/datum/element/drag_pickup)
-	if(!mapload)
-		return
-	var/obj/item/pen/P = locate(/obj/item/pen) in src.loc
-	if(P && !bin_pen)
-		P.forceMove(src)
-		bin_pen = P
-		update_appearance()
-	
+	if(mapload)
+		var/obj/item/pen/pen = locate(/obj/item/pen) in loc
+		if(pen && !bin_pen)
+			pen.forceMove(src)
+			bin_pen = pen
+	for(var/i in 1 to total_paper)
+		papers.Add(generate_paper())
+	update_appearance()
+
+/obj/item/paper_bin/proc/generate_paper()
+	var/obj/item/paper/paper = new papertype(src)
+	if(SSevents.holidays && SSevents.holidays[APRIL_FOOLS])
+		if(prob(30))
+			paper.info = "<font face=\"[CRAYON_FONT]\" color=\"red\"><b>HONK HONK HONK HONK HONK HONK HONK<br>HOOOOOOOOOOOOOOOOOOOOOONK<br>APRIL FOOLS</b></font>"
+			paper.AddComponent(/datum/component/honkspam)
+	return paper
 
 /obj/item/paper_bin/Destroy()
-	if(papers)
-		for(var/i in papers)
-			qdel(i)
-		papers = null
+	if(LAZYLEN(papers))
+		for(var/paper in papers)
+			qdel(paper)
+		LAZYNULL(papers)
```

Comment:
```suggestion
	QDEL_LIST(papers)
```

Works with lazy lists too.

Reply:
Thanks, I like to learn things like this